hal/src/nRF52840/ble_hal.cpp

```diff
@@ -327,7 +327,7 @@ class BleObject::Broadcaster {
     volatile hal_ble_auto_adv_cfg_t autoAdvCfg_;    /**< Automatic advertising configuration. */
     uint8_t advHandle_;                             /**< Advertising handle. */
     hal_ble_adv_params_t advParams_;                /**< Current advertising parameters. */
-    uint8_t advData_[BLE_MAX_ADV_DATA_LEN];         /**< Current advertising data. */
+    uint8_t advData_[BLE_MAX_SCAN_REPORT_BUF_LEN];  /**< Current advertising data. */
```

Comment:
I'd suggest to introduce dedicate macros for adv data, e.g. 
```
#define BLE_MAX_ADV_DATA_LEN_EXT BLE_GAP_ADV_SET_DATA_SIZE_EXTENDED_MAX_SUPPORTED
#define BLE_MAX_ADV_DATA_LEN_EXT_CONNECTABLE BLE_GAP_ADV_SET_DATA_SIZE_EXTENDED_CONNECTABLE_MAX_SUPPORTED
```
and use `BLE_MAX_ADV_DATA_LEN_EXT` for general adv data buffer, while use `BLE_MAX_SCAN_REPORT_BUF_LEN` for scan report buffer.

Reply:
👍  Added